tests/core/decorator.test.ts

```diff
@@ -8,4 +19,178 @@ describe(`Decorators`, () => {
         
         expect(getSerializationMetadata(Foo)?.uniquelyReferenced).toBe(true);
     });
+
+    @ccclass('Dummy')
+    class Dummy {}
+
+    describe('Visible', () => {
+        function visibleFn () { return false; }
+
+        const withVisibleTrueSerializableFalse = <T>(v: T) => ({ visible: true, serializable: false, ...v });
+
+        const withVisibleFalseSerializableTrue = <T>(v: T) => ({ visible: false, serializable: true, ...v });
+
+        test.each([
+            ['@property', [() => property], { /*visible: true, serializable: true*/ }],
```

Comment:
`[test title, [decorators applyied], the result of CCClass.Attr.getAttrs() on this property]`